clerk/domains.go

```diff
@@ -0,0 +1,80 @@
+package clerk
+
+import (
+	"fmt"
+)
+
+type DomainsService service
+
+type DomainListResponse struct {
+	Data       []Domain `json:"data"`
+	TotalCount int      `json:"total_count"`
+}
+
+type Domain struct {
+	Object   string `json:"object"`
+	ID       string `json:"id"`
+	Name     string `json:"name"`
+	ProxyURL string `json:"proxy_url"`
+}
+
+type CreateDomainParams struct {
+	Name        string  `json:"name"`
+	IsSatellite bool    `json:"is_satellite"`
+	ProxyURL    *string `json:"proxy_url,omitempty"`
+}
+
+type UpdateDomainParams struct {
+	Name     *string `json:"name,omitempty"`
+	ProxyURL *string `json:"proxy_url,omitempty"`
+}
+
+func (s *DomainsService) ListAll() ([]Domain, error) {
+	req, _ := s.client.NewRequest("GET", DomainsURL)
```

Comment:
:upside_down_face: 
```suggestion
	req, _ := s.client.NewRequest(http.MethodGet, DomainsURL)
```
Same for all of the methods.